wodles/docker-listener/DockerListener.py

```diff
@@ -0,0 +1,139 @@
+        self.wazuh_queue = '{0}/queue/ossec/queue'.format(self.wazuh_path)
+        self.msg_header = "1:Wazuh-Docker:"
+        # docker variables
+        self.client = docker.from_env()
+        self.send_msg(json.dumps({self.field_debug_name: "Started"}))
+        self.thread1 = threading.Thread(target=self.listen)
+        self.thread2 = threading.Thread(target=self.listen)
+        self.connect(first_time=True)
+
+    def connect(self, first_time=False):
+        """
+        Connects to Docker service, retrying if connection is not successful
+
+        """
+        if self.check_docker_service():
+            if not first_time:
+                # this is for having a thread assigned to a variable ever
+                if self.thread1.is_alive():
+                    self.thread2 = threading.Thread(target=self.listen)
+                    self.thread2.start()
+                else:
+                    self.thread1 = threading.Thread(target=self.listen)
+                    self.thread1.start()
+            else:
+                self.thread1.start()
+            print("Docker service was started.")
+            self.send_msg(json.dumps({self.field_debug_name: "Connected to Docker service"}))
+        else:
+            if first_time:
+                print("Docker service is not running.")
+                self.send_msg(json.dumps({self.field_debug_name: "Docker service is not running"}))
+            while not self.check_docker_service():
+                print("Reconnecting...")
+                time.sleep(self.wait_time)
+            self.connect()
+
+    def check_docker_service(self):
+        """
+        Checks if Docker service is running
+
+        :return: True if Docker service is active or False if it is inactive.
+        """
+        try:
+            self.client.ping()
+            return True
+        except Exception:
+            return False
+
+    def listen(self):
+        """
+        Listens Docker events
+
+        """
+        try:
+            for event in self.client.events():
+                self.process(event)
+        except Exception:
+            raise Exception
+        print("Docker service was stopped.")
+        self.send_msg(json.dumps({self.field_debug_name: "Disconnected from the Docker service"}))
+        self.connect()
+
+    def process(self, event):
+        """"
+        Processes a main Docker event
+
+        :param event: Docker event.
+        """
+        self.send_msg(event.decode("utf-8"))
+
+    def format_msg(self, msg):
+        """
+        Formats a Docker event
+
+        :param msg: message to be formatted.
+        :return: formatted message.
+        """
+        return {'integration': 'docker', 'docker': json.loads(msg)}
+
+    def send_msg(self, msg):
+        """
+        Sends a Docker event to the Wazuh Queue
+
+        :param msg: message to be sent.
+        """
+        try:
+            json_msg = json.dumps(self.format_msg(msg))
+            print(json_msg)
```

Comment:
This can be pretty noisy if lots of events are generated. Wouldn't it better to print this message only if debugging is enabled?